Uploading & Downloading Benchmarking Data

shreyasingh1 opened the pull request:
Created upload_benchmarking.py to automate scaling and upload of the 50 image benchmarking data and their associated swc files.  Made changes to swc.py to automate this process as well. Created uploading notebook called upload_benchmarking.ipynb. Upload destination is: s3://open-neurodata/brainlit/benchmarking_data.

Created download_benchmarking.ipynb tutorial to show (1) how to download the data from S3 using Neuroglancer and (2) how to locally download the data using the AWS CLI. To automate download, a few minor changes were made in session.py and swc.py in the form of optional parameters to existing functions. The downloading notebook can visualize any .tif file and its overlaid .swc's, either from S3 or from a local file location.

Commit:
first commit for benchmarking_data.ipynb notebook, loadeed benchmarking
data locally, waiting on script to produce ground truth data before
continuing with preprocessing

Commit:
started creating notebook (adapted from
generate_dataset_tube_anisotropic.py in the mouselight_code repo)

Commit:
updated loading benchmarking data. currently having problems when
calling read_swc.py (in mouselight_code repo)

Commit:
altered indeces of the file name parser

shreyasingh1 committed:
Create upload_masks.py

shreyasingh1 committed:
Update upload_masks.py

shreyasingh1 committed:
Update __init__.py

Commit:
creating upload_benchmarking.py, which is a copy of upload.py with
slight changes, to automate uploading the
benchmarking data

Commit:
added upload_benchmarking.py to __init__ file

Commit:
shreya updates to uploading_brains to automate uploading the
benchmarking data

Commit:
renamed

Commit:
finished uploading swc files

Commit:
Cleaning up benchmarking data upload steps & detailing timeout error

Commit:
reverted back to original uploading_brains. Added prev commits to
uploading_benchmarking.ipynb

Commit:
process for uploading benchmarking data using upload_benchmarking.py

Commit:
update
 Please enter the commit message for your changes. Lines starting

Commit:
uploading benchmarking data in octree format. modifying idx in swc.py

FelixChihTing committed:
simplify path-searching process

FelixChihTing committed:
Merge branch 'develop' of https://github.com/NeuroDataDesign/brainlit into CTSprint2

Commit:
updated downloading benchmarking ease and steps

Commit:
created downloading_brains tutorial notebook

Commit:
updating/cleaning

Commit:
updating

FelixChihTing committed:
DataFrame as input to GeometriGraph class

Commit:
reverted to original

Commit:
cleaned & black formatted

Commit:
cleaned

Commit:
clearing cell output

Commit:
updated

tathey1 committed:
Update README.md

tathey1 committed:
Merge branch 'develop' into tathey1-readme

bvarjavand committed:
Merge without updating requirements

bvarjavand committed:
Add requirements fixes and wincertstore

bvarjavand committed:
Merge branch 'develop' into netlify-fix

bvarjavand committed:
reinclude code so tests pass

bvarjavand committed:
Merge branch 'netlify-fix' of https://github.com/NeuroDataDesign/brainlit into netlify-fix

bvarjavand committed:
merge

bvarjavand committed:
netlify fixes

bvarjavand committed:
comment out code which was commented in PR

bvarjavand committed:
Merge pull request #174 from NeuroDataDesign/netlify-fix

Merge with netlify fixes.

bvarjavand committed:
Merge branch 'develop' into tathey1-readme

tathey1 committed:
Merge pull request #173 from neurodata/tathey1-readme

Update README.md to address several items here: https://github.com/microsoft/graspologic/issues/531

Namely, adds code of conduct, names of maintainers, and removes other things that can be found in our documentation site.

Commit:
finalized, formatted

Commit:
removing old files

Commit:
removed dependence on mouselight_code

Commit:
minor changes for build

Commit:
allowing optional parameters in swc.py

tathey1 commented on brainlit/utils/session.py:
```diff
@@ -94,6 +94,7 @@ def _get_voxel(self, seg_id: int, v_id: int) -> Tuple[int, int, int]:
+        voxel = vertex.astype(int)
         voxel = np.round(
             np.divide(vertex, self.cv_segments.scales[self.mip]["resolution"])
         ).astype(int)
```
You seem to redefine voxel immediately - isn't line 97 useless?

shreyasingh1 replied:
Yep, thanks for catching that. Removed!

tathey1 commented on brainlit/utils/session.py:
```diff
@@ -110,12 +111,18 @@ def set_url_segments(self, seg_url: str):
         Arguments:
             seg_id  The segement to pull.
             bbox: The bounding box object, default None. If None, uses entire volume.
+            benchmark: Optional, default False. If True (relevant for benchmarking data), S3 file isn't rounded.
```
I'd like to avoid having a whole extra argument for just this dataset if possible. Is it possible to remove rounding from the original function? And then if there are other things that depend on the rounding just make those things round themselves?

tathey1 commented on brainlit/utils/swc.py:
```diff
@@ -174,7 +175,7 @@ def bbox_vox(df):
 
 
-def read_s3(s3_path, seg_id, mip):
+def read_s3(s3_path, seg_id, mip, benchmark: Optional[bool] = False):
```
same comment as above - can you try to remove this argument?

tathey1 commented on brainlit/utils/swc.py:
```diff
@@ -215,10 +218,16 @@ def read_s3(s3_path, seg_id, mip):
+        df["x"] = np.round(df["x"] / res[0])
+        df["y"] = np.round(df["y"] / res[1])
+        df["z"] = np.round(df["z"] / res[2])
+
```
like earlier comment - if you remove this code will it break anything? worth looking into

tathey1 reviewed (commented):
The notebooks are beautiful.

My main question is why did you need to write a whole new upload_benchmarking.py? Why couldn't you have reused code from upload.py? Not saying it's not ok, but I want to know. Can @bvarjavand review upload_benchmarking.py and download_benchmarking.py and comment on whether its possible to combine with upload/py/download.py?

tathey1 reviewed (changes requested):
a few minor comments, and a discussion id like to have with Bijan

bvarjavand commented:
It seems like the main difference is the point at which you scale the data. I don't see any obvious reason why you couldn't merge this with upload.py (maybe you thought you shouldn't edit existing code? Then our bad, you have the go-ahead as long as tests pass).

shreyasingh1 commented:
@tathey1 The reason I kept upload_benchmarking.py separate is because I didn't want to break anything in upload.py, but I do now think that combining the scripts would make upload.py a bit heavy. 
Specifically:
- in get_volume_info I imported the benchmarking scaling information, calculated offset and used that to set the origin, and also modified the file path line and img_size line
- in create_cloud_volume I made the chunk size 1 instead of 4 and changed the scaling
- in get_data_ranges i modified the scale factor
- in upload_segments i added a line to divide vertices by the swc scaling factor
By contrast, in swc.py I only needed to make small changes so I just made a new function instead of making an entire swc_benchmarking.py. That being said, happy to try to combine them if you all think that's best.

In regards to the benchmarking parameter, I used it to just get around the rounding step in read_s3. When there's rounding, the benchmarking swc files load incorrectly (just as 0s and 1s) but without rounding they load correctly. Without rounding, the regular downloading_brains notebook also works. @bvarjavand do you know of anywhere else where rounding is necessary? If there isn't too many, I can remove the benchmarking parameter. If the rounding part is important though, is there a better way to get around it than an Optional benchmarking parameter?

tathey1 commented:
Regarding rounding, from my perspective (worth getting Bijan's also), as long as things don't break, nothing is inherently off limits.

Regarding the upload scripts, I think it's definitely worth trying to use the original create_cloud_volume and get_data_ranges functions, just add an argument to them for chunk size/scale factor and you should be good. If this is possible for get_volume_info also, I think it's worth it.

Do you think you can combine those files so there's only a single upload.py and no upload_benchmarking.py? Is there anything else besides using keyword arguments that you'd need to do to combine?

Commit:
merging upload_benchmarking.py into upload.py

Commit:
updating init file

shreyasingh1 commented:
@tathey1 just merged upload_benchmarking.py into upload.py and both upload_brains.ipynb and upload_benchmarking.ipynb still run! I also changed the name of the benchmarking parameter to "rounding" to make it less specific to the benchmarking data, but can get rid of it all together as well.

FelixChihTing committed:
remove session 2

FelixChihTing committed:
Merge branch 'develop' of https://github.com/NeuroDataDesign/brainlit into CTSprint2

FelixChihTing committed:
give path to swc instead of searching by function

FelixChihTing committed:
define variables in the same code block

tathey1 commented:
@shreyasingh1 there are some merge conflicts with this PR, please merge your branch with deploy. Other than that, I think things look good.

shreyasingh1 committed:
Merge branch 'develop' into shreya_dn

shreyasingh1 commented:
@tathey1 sounds good, just addressed the merge conflicts!

bvarjavand commented on brainlit/utils/swc.py:
```diff
@@ -186,6 +187,8 @@ def read_s3(s3_path, seg_id, mip):
         The segement number to pull
     mip : int
         The resolution to use for scaling
+    benchmark: bool, Optional
```
change this to the correct parameter name

bvarjavand commented on brainlit/utils/swc.py:
```diff
@@ -512,3 +522,51 @@ def swc2skeleton(swc_file, origin=None):
     return skel
+
+
+def swc2skeleton_benchmarking(swc_file, origin=None):
```
it looks like the only difference between this method and `swc2skeleton` method is the way you get the ID and `origin` not being checked before subtraction. Seems reasonable to just update the existing method, why make a new one?

bvarjavand commented on docs/notebooks/utils/downloading_brains.ipynb:
```diff
@@ -131,5 +110,6 @@
   }
+
  ]
-}
```
you deleted the entire downloading_brains notebook? Is this necessary for your PR to pass?

bvarjavand reviewed (commented):
Looks good. I currently don't see any output for your notebooks when I look at your branch. How do I know they run without issue?

tathey1 commented:
Then please open this PR into [here](https://github.com/neurodata/brainlit), and close this one

FelixChihTing committed:
use pathlib to locate direct path to swc

FelixChihTing committed:
remove for loop when defining soma

FelixChihTing committed:
remove comments

bvarjavand committed:
Merge branch 'develop' into CTSprint2

tathey1 committed:
Merge pull request #178 from neurodata/CTSprint2

Ct sprint2

shreyasingh1 committed:
Merge branch 'develop' into shreya_dn

Commit:
removing swc2skeleton_benchmarking, adding notebook output

Commit:
Merge branch 'shreya_dn' of https://github.com/NeuroDataDesign/brainlit into shreya_dn
